check/type-definition.spec.ts

```diff
@@ -23,6 +24,15 @@ result.formatWith(error => {
   error.value;
 });
 
+withDefaults({
```

Comment:
Can you use `validationResult.withDefaults()`? So that it matches the documented example.
I think this will also require changes to how the type declaration is done...

Reply:
It will also be worth changing the test to `const result: ValidationResult = validationResult.withDefaults(...)(req)`. 
Should catch all possible typing bugs...

Reply:
I think it should be set now.

I kept `validationResult` interface as lower-camel-cased, since it might be more easily identifiable as a function. If any objections, I can also change to `ValidationResult`.